rest-endpoint/src/main/java/com/cognifide/aet/rest/MetadataServlet.java

```diff
@@ -35,19 +35,18 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import org.apache.commons.lang3.BooleanUtils;
-import org.apache.felix.scr.annotations.Activate;
-import org.apache.felix.scr.annotations.Component;
-import org.apache.felix.scr.annotations.Deactivate;
-import org.apache.felix.scr.annotations.Reference;
-import org.apache.felix.scr.annotations.Service;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.http.HttpService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-@Service
-@Component(label = "MetadataServlet", description = "Returns Suite Metadata", immediate = true)
+@Component(service = BasicDataServlet.class, immediate = true)
```

Comment:
Could you check if `service = HttpServlet.class` would be enough here and in other `BasicDataServlet` extensions please? We only want this class to be registered as a servlet. It won't be referenced anywhere.